src/Input/Keybindings.re

```diff
@@ -111,22 +158,117 @@ module Legacy = {
     };
 };
 
-let of_yojson_with_errors = json => {
-  switch (json) {
-  // Current format:
-  // [ ...bindings ]
-  | `List(bindingsJson) =>
-    let res = bindingsJson |> Internal.of_yojson_with_errors;
-    Ok(res);
-  // Legacy format:
-  // { bindings: [ ..bindings. ] }
-  | `Assoc([("bindings", `List(bindingsJson))]) =>
-    let res = bindingsJson |> Internal.of_yojson_with_errors;
-
-    Ok(res)
-    |> Stdlib.Result.map(((bindings, errors)) => {
-         (Legacy.upgrade(bindings), errors)
-       });
-  | _ => Error("Unable to parse keybindings - not a JSON array.")
+let keyToSdl =
+  EditorInput.Key.(
+    {
+      fun
+      | Escape => "Escape"
+      | Return => "Return"
+      | Up => "Up"
+      | Down => "Down"
+      | Left => "Left"
+      | Right => "Right"
+      | Tab => "Tab"
+      | PageUp => "PageUp"
+      | PageDown => "PageDown"
+      | Delete => "Delete"
+      | Pause => "Pause"
+      | Home => "Home"
+      | End => "End"
+      | Backspace => "Backspace"
+      | CapsLock => "CapsLock"
+      | Insert => "Insert"
+      | Function(digit) => "F" ++ string_of_int(digit)
+      | Space => "Space"
+      | NumpadDigit(digit) => "Keypad " ++ string_of_int(digit)
+      | NumpadMultiply => "Keypad *"
+      | NumpadAdd => "Keypad +"
+      | NumpadSeparator => "Keypad "
+      | NumpadSubtract => "Keypad -"
+      | NumpadDivide => "Keypad //"
+      | NumpadDecimal => "Keypad ."
+      | Character(c) => String.make(1, c) |> String.uppercase_ascii;
+    }
+  );
+
+let codeToOpt =
+  fun
+  | 0 => None
+  | x => Some(x);
+
+let getKeycode = keycodeStr => {
+  keycodeStr |> keyToSdl |> Sdl2.Keycode.ofName |> codeToOpt;
+};
+
+let getScancode = scancodeStr => {
+  scancodeStr |> keyToSdl |> Sdl2.Scancode.ofName |> codeToOpt;
+};
```

Comment:
`keycodeStr` and `scancodeStr` does not seem to be strings, as `keyToSdl` takes a variant.

Reply:
Ah, ya, prior to https://github.com/onivim/editor-input/pull/6 they were `strings`, but missed updating these names once they came in. Fixed